### ISSUE TITLE
classifier loss coming out to be NaN

### ISSUE THREAD

**Issue by letdivedeep** [sayakpaul/Supervised-Contrastive-Learning-in-TensorFlow-2#6]
Hi @sayakpaul 

thanks for this wonderful work

I ran the model on the 5 class imageNet dataset it works like a charm there on my local infra.

for one of my custom model with ard 40 classes, the encoder trains well but the classifier loss is showing as nan itself.

![Screenshot 2021-04-05 at 8 58 07 PM](https://user-images.githubusercontent.com/74127861/113591567-ab933200-9651-11eb-9375-4d9a541835bb.png)

**Comment by sayakpaul** [comment]
Maybe additional hyperparameter tuning and regularization are required to stabilize the training. For regularization, I would suggest the following:

* L2 weight decay with a factor or 0.005.
* BatchNorm in the projection head to ease the optimization. This has a regularization effect as well
* Use RandAugment in phase 1 of the training. [Here's an example](https://keras.io/examples/vision/randaugment/) of how to use them alongside Keras. 

Also, if your dataset is larger I would suggest trying out a deeper variant of ResNet, ResNet101, for example.

**Comment by letdivedeep** [comment]
@sayakpaul  thanks for the reply 

I will try out the above-listed suggestion. and get back to you

**Comment by letdivedeep** [comment]
Hi @sayakpaul,

regularization did the trick. 

just can we have some sort of hack ard with this array bcoz when we increase the dataset count it comes as memory full.

**Comment by sayakpaul** [comment]
Will suggest checking out `tf.data` guidelines here: https://www.tensorflow.org/guide/data_performance.
